docs/dotnet/dotnet-http-metrics.md

```diff
@@ -0,0 +1,383 @@
+
+If the HTTP request method is not known to instrumentation, it MUST set the `http.request.method` attribute to `_OTHER` and, except if reporting a metric, MUST
+set the exact method received in the request line as value of the `http.request.method_original` attribute.
+
+If the HTTP instrumentation could end up converting valid HTTP request methods to `_OTHER`, then it MUST provide a way to override
+the list of known HTTP methods. If this override is done via environment variable, then the environment variable MUST be named
+OTEL_INSTRUMENTATION_HTTP_KNOWN_METHODS and support a comma-separated list of case-sensitive known HTTP methods
+(this list MUST be a full override of the default known method, it is not a list of known methods in addition to the defaults).
+
+HTTP method names are case-sensitive and `http.request.method` attribute value MUST match a known HTTP method name exactly.
+Instrumentations for specific web frameworks that consider HTTP methods to be case insensitive, SHOULD populate a canonical equivalent.
+Tracing instrumentations that do so, MUST also set `http.request.method_original` to the original value.
+
+**[2]:** `network.protocol.version` refers to the version of the protocol used and might be different from the protocol client's version. If the HTTP client used has a version of `0.27.2`, but sends HTTP version `1.1`, this attribute should be set to `1.1`.
+
+`http.request.method` has the following list of well-known values. If one of them applies, then the respective value MUST be used, otherwise a custom value MAY be used.
+
+| Value  | Description |
+|---|---|
+| `CONNECT` | CONNECT method. |
+| `DELETE` | DELETE method. |
+| `GET` | GET method. |
+| `HEAD` | HEAD method. |
+| `OPTIONS` | OPTIONS method. |
+| `PATCH` | PATCH method. |
+| `POST` | POST method. |
+| `PUT` | PUT method. |
+| `TRACE` | TRACE method. |
+| `_OTHER` | Any HTTP method that the instrumentation has no prior knowledge of. |
+<!-- endsemconv -->
+
+## HTTP server
+
+All Http server metrics are reported by `Microsoft.AspNetCore.Hosting` meter.
+
+### Metric: `http.server.duration`
+
+<!-- semconv metric.dotnet.http.server.duration(metric_table) -->
+| Name     | Instrument Type | Unit (UCUM) | Description    |
+| -------- | --------------- | ----------- | -------------- |
+| `http.server.duration` | Histogram | `s` | Measures the duration of inbound HTTP requests. |
+<!-- endsemconv -->
+
+<!-- semconv metric.dotnet.http.server.duration(full) -->
+| Attribute  | Type | Description  | Examples  | Requirement Level |
+|---|---|---|---|---|
+| `aspnet.request.is_handled` | boolean | Flag indicating if request was handled by the application pipeline. TODO | `True`; `False` | Recommended |
+| `http.error.reason` | string | Low-cardinality error reason: `timeout`, `cancellation` or TODO(link to possible errors) [1] | `timeout`; `cancellation`; `client_closed_connection-todo` | Conditionally Required: If request has failed. |
```

Comment:
makes sense, as long as we don't expect errors that are more granular than exceptions (as in the case of client errors).

Reply:
I thought in our earlier discussion @lmolkova your goal for `http.error.reason` was to have a standardized attribute name. The description in the text was misleading, but when I talked with @lmolkova about it she said it could be used as a free-form string for implementations to put whatever they wanted. ASP.NET would use it for exception types.

I have no objection to returning the attribute to `exception.type.` I think that was my original suggestion for it.

Reply:
1. we can do `exception.type` if we believe exception type is the best we'd ever need on asp.net core.

2. an alternative: `http.error.reason`  that has 'cancellation', 'timeout', or exception type if it's not `OperationCancelledException` or `TimeoutException`. Then we'd have consistent attributes across client and server, consistent value for cancellation (and maybe other classes of errors in the future). 

I lean to `http.error.reason`, but would be fine with `exception.type` on the server